devicetypes/smartthings/zigbee-sound-sensor/zigbee-sound-sensor.groovy

```diff
@@ -0,0 +1,175 @@
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+        valueTile("temperature", "device.temperature", width: 2, height: 2) {
+            state("temperature", label: '${currentValue}°',
+                    backgroundColors: [
+                            [value: 31, color: "#153591"],
+                            [value: 44, color: "#1e9cbb"],
+                            [value: 59, color: "#90d2a7"],
+                            [value: 74, color: "#44b621"],
+                            [value: 84, color: "#f1d801"],
+                            [value: 95, color: "#d04e00"],
+                            [value: 96, color: "#bc2323"]
+                    ])
+        }
+
+        main "sound"
+        details(["sound", "battery", "temperature"])
+    }
+}
+
+private getPOLL_CONTROL_CLUSTER() { 0x0020 }
+private getFAST_POLL_TIMEOUT_ATTR() { 0x0003 }
+private getCHECK_IN_INTERVAL_ATTR() { 0x0000 }
+private getBATTERY_VOLTAGE_VALUE() { 0x0020 }
+private getTEMPERATURE_MEASURE_VALUE() { 0x0000 }
+private getSET_LONG_POLL_INTERVAL_CMD() { 0x02 }
+private getSET_SHORT_POLL_INTERVAL_CMD() { 0x03 }
+private getCHECK_IN_INTERVAL_CMD() { 0x00 }
+
+def installed() {
+    sendEvent(name: "sound", value: "not detected", displayed: false)
+}
+
+def parse(String description) {
+    def map = zigbee.getEvent(description)
+
+    if(!map) {
+        if(description?.startsWith('zone status')) {
+            map = parseIasMessage(description)
+        } else {
+            map = parseAttrMessage(description)
+        }
+    }
+
+    def result = map ? createEvent(map) : [:]
+
+    if (description?.startsWith('enroll request')) {
+        def cmds = zigbee.enrollResponse()
+        log.debug "enroll response: ${cmds}"
+        result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+    }
+    return result
+}
+
+private Map parseIasMessage(String description) {
+    ZoneStatus zs = zigbee.parseZoneStatus(description)
+    def result = [:]
+    if(zs.isAlarm1Set() || zs.isAlarm2Set()) {
+        result = getSoundDetectionResult("detected")
+    } else if(!zs.isTamperSet()) {
+        result = getSoundDetectionResult("not detected")
+    } else {
+        result = [displayed: true, descriptionText: "${device.displayName}'s case is opened"]
+    }
+
+    return result
+}
+
+private Map parseAttrMessage(description) {
+    def descMap = zigbee.parseDescriptionAsMap(description)
+    def map = [:]
+    if(descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+        map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+    } else if(descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+        if (descMap.data[0] == "00") {
+            sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```

Comment:
Extract  sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee"... to a method. You will not need to change it in two places if required.